pyro/poutine/__init__.py

```diff
@@ -99,14 +98,32 @@ def escape(fn, escape_fn=None):
     and returns a boolean value to decide whether to exit at that site
     :returns: stochastic function wrapped in EscapePoutine
 
-    Alias for EscapePoutine constructor.
+    Alias for ContinuationPoutine constructor.
+    Formerly an alias for EscapePoutine constructor.
 
     Given a callable that contains Pyro primitive calls,
     evaluate escape_fn on each site, and if the result is True,
     raise a NonlocalExit exception that stops execution
     and returns the offending site.
     """
-    return EscapePoutine(fn, escape_fn)
+    return ContinuationPoutine(fn, escape_fn, util.escape_cont_fn)
+
+
+def enum(fn, first_available_dim):
```

Comment:
Do we really want to expose this? I'm tempted to move all the enumeration-related helpers to pyro.infer.enum. Since that library is undocumented (i.e. absent from sphinx), we can reduce guarantees of backwards compatibility.

Reply:
This is just for API compatibility during the refactor, will remove later